### PR TITLE
Torch original unet

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
This PR implements the original unet for Noise2Self.

### CODE REVIEW

File: aydin/nn/layers/custom_conv.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            self.instance_normalization = nn.InstanceNorm2d(out_channels)
+            self.batch_normalization = nn.BatchNorm2d(out_channels, affine=False)
+        else:
+            self.conv = nn.Conv3d(
+                in_channels, out_channels, kernel_size, padding='same'
+            )
+            self.instance_normalization = nn.InstanceNorm3d(out_channels)
+            self.batch_normalization = nn.BatchNorm3d(out_channels, affine=False)
+
+        self.activation_function = {
```
Comment (li-li-github) [inline aydin/nn/layers/custom_conv.py:36]: this semantic looks nice

File: aydin/nn/layers/custom_conv.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    def forward(self, x):
+        x = self.conv(x)
+
+        if self.normalization == 'instance':
+            x = self.instance_normalization(x)
+        elif self.normalization == 'batch':
+            x = self.batch_normalization(x)
+
+        x = self.activation_function(x)
```
Comment (li-li-github) [inline aydin/nn/layers/custom_conv.py:50]: If none of the correct activation key was passed, will this become `x = None(x)` ?

File: aydin/nn/models/torch/test/test_torch_unet.py [hunk 3]
```diff
@@ -45,27 +32,32 @@ def test_supervised_2D_n2t():
...
+
+        napari.run()
 
     # assert result.shape == input_image.shape
     # assert result.dtype == input_image.dtype
 
 
-def test_masking_2D():
-    input_array = torch.zeros((1, 1, 64, 64))
+@pytest.mark.parametrize("nb_unet_levels", [2, 3, 5, 8])
```
Comment (li-li-github) [inline aydin/nn/models/torch/test/test_torch_unet.py:55]: 8 layers?!

File: aydin/nn/layers/custom_conv.py (hunk 2 above)
Comment (AhmetCanSolak) [reply aydin/nn/layers/custom_conv.py:50]: no, it will give error within the __init__

File: aydin/nn/models/torch/test/test_torch_unet.py (hunk 3 above)
Comment (AhmetCanSolak) [reply aydin/nn/models/torch/test/test_torch_unet.py:55]: Might worth to test, don't you think so? :)

File: aydin/nn/layers/custom_conv.py (hunk 2 above)
Comment (li-li-github) [reply aydin/nn/layers/custom_conv.py:50]: We might want to make an option not to get any activation layer just like the normalization layer.

**Review by li-li-github — COMMENTED** [review]
Overall it looks good to me!
